hubblestack/daemon.py

```diff
@@ -466,6 +469,45 @@ def parse_args():
                         help='Any arguments necessary for a single function run')
     return vars(parser.parse_args())
 
+def check_pidfile(kill_other=False):
+    '''
+    Check to see if there's already a pidfile. If so, check to see if the
+    indicated process is alive and is Hubble.
+
+    kill_other
+        Default false, if set to true, attempt to kill detected running Hubble
+        processes; otherwise exit with an error.
+
+    '''
+    pidfile = __opts__['pidfile']
+    if os.path.isfile(pidfile):
```

Comment:
I think a concern was surfaced elsewhere along the lines that, what if a pid file is missing, would we still get a duplicate process?

Reply:
Probably. I'm not aware of many daemons that check to see if another instances is running without using a pidfile like this. Databases tend to do it, because they notice eachother accessing the same database files. That's just about it though.

I could certainly add further checks, but crawling through the process list for things that happen to be named hubble might result as many false positives as not. Another way to approach it might be to keep a lock on some file in /var/lib/hubble or something. But older versions of hubble wouldn't have that lock, so it wouldn't help there either.

Reply:
/proc/<pid\>/exe seems to point to _/opt/hubble/hubble-libs/hubble_ even in older versions.
This needs a case when a process is there but the file has been deleted during the upgrade. It should still have a pointer, even when it says deleted.

Given the possible slow adoption schedule for new versions I think it's a good idea to fix this problem so that is unlikely to occur again.